src/thor/triplegbuilder.cc

```diff
@@ -462,6 +462,285 @@ void SetHeadings(TripLeg_Edge* trip_edge,
+    traversability = (intersecting_de->reverseaccess() & kPedestrianAccess)
+                         ? Traversability::kBoth
+                         : Traversability::kForward;
+  } else {
+    traversability = (intersecting_de->reverseaccess() & kPedestrianAccess)
+                         ? Traversability::kBackward
+                         : Traversability::kNone;
+  }
+  // Set the walkability flag for the intersecting edge if requested
+  if (controller.attributes.at(kNodeIntersectingEdgeWalkability)) {
+    intersecting_edge->set_walkability(GetTripLegTraversability(traversability));
+  }
+
+  traversability = Traversability::kNone;
+  // Determine cyclability
+  if (intersecting_de->forwardaccess() & kBicycleAccess) {
+    traversability = (intersecting_de->reverseaccess() & kBicycleAccess) ? Traversability::kBoth
+                                                                         : Traversability::kForward;
+  } else {
+    traversability = (intersecting_de->reverseaccess() & kBicycleAccess) ? Traversability::kBackward
+                                                                         : Traversability::kNone;
+  }
+  // Set the cyclability flag for the intersecting edge if requested
+  if (controller.attributes.at(kNodeIntersectingEdgeCyclability)) {
+    intersecting_edge->set_cyclability(GetTripLegTraversability(traversability));
+  }
+
+  // Set the driveability flag for the intersecting edge if requested
+  if (controller.attributes.at(kNodeIntersectingEdgeDriveability)) {
+    intersecting_edge->set_driveability(
+        GetTripLegTraversability(nodeinfo->local_driveability(local_edge_index)));
+  }
+
+  // Set the previous/intersecting edge name consistency if requested
+  if (controller.attributes.at(kNodeIntersectingEdgeFromEdgeNameConsistency)) {
+    bool name_consistency =
+        (prev_de == nullptr) ? false : prev_de->name_consistency(local_edge_index);
+    intersecting_edge->set_prev_name_consistency(name_consistency);
+  }
+
+  // Set the current/intersecting edge name consistency if requested
+  if (controller.attributes.at(kNodeIntersectingEdgeToEdgeNameConsistency)) {
+    intersecting_edge->set_curr_name_consistency(directededge->name_consistency(local_edge_index));
+  }
+
+  // Set the use for the intersecting edge if requested
+  if (controller.attributes.at(kNodeIntersectingEdgeUse)) {
+    intersecting_edge->set_use(GetTripLegUse(intersecting_de->use()));
+  }
+
+  // Set the road class for the intersecting edge if requested
+  if (controller.attributes.at(kNodeIntersectingEdgeRoadClass)) {
+    intersecting_edge->set_road_class(GetRoadClass(intersecting_de->classification()));
+  }
+
+  // Set the lane count for the intersecting edge if requested
+  if (controller.attributes.at(kNodeIntersectingEdgeLaneCount)) {
+    intersecting_edge->set_lane_count(intersecting_de->lanecount());
+  }
+
+  // Set the sign info for the intersecting edge if requested
```

Comment:
Here's some new logic.  This gets the signs from the `intersecting_de` and puts them on the `valhalla::TripSign`.